pkg/daemon/update.go

```diff
@@ -579,15 +579,10 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		return err
 	}
 
-	defer func() {
-		if retErr != nil {
-			if err := dn.updateSSHKeys(oldIgnConfig.Passwd.Users); err != nil {
-				errs := kubeErrs.NewAggregate([]error{err, retErr})
-				retErr = fmt.Errorf("error rolling back SSH keys updates: %w", errs)
-				return
-			}
-		}
-	}()
```

Comment:
I think this defer got accidentally removed -- we do need it, and we also will need a matching one for the password so the password gets reverted if we hit an error.

Reply:
yes i accidentally removed this- thanks john!